httpsig.go

```diff
@@ -225,6 +254,31 @@ func NewResponseVerifier(r *http.Response) (Verifier, error) {
 	})
 }
 
+func newSignerFromSSHSigner(sshSigner ssh.Signer, algo Algorithm, dAlgo DigestAlgorithm, headers []string, scheme SignatureScheme, expiresIn int64) (Signer, error) {
```

Comment:
nit: let's rename `newSignerFromSSHSigner` to `newSSHSigner` if we're introducing a `SSHSigner` interface (see above comment)